ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.revenuecat.purchases.ui.revenuecatui.data.processed
+
+import android.content.Context
+import com.revenuecat.purchases.Package
+import java.text.NumberFormat
+import java.util.Currency
+import java.util.Locale
+
+@Suppress("UnusedParameter", "FunctionOnlyReturningConstant")
+internal object VariableDataProvider {
+    private const val MICRO_MULTIPLIER = 1000000.0
+
+    fun applicationName(context: Context): String {
```

Comment:
Oh I see you already have this. Instead of making `context` a parameter of this, we could add an `ApplicationContext` interface (and make `Context` implement it), so this depends on only a tiny interface instead of a full object.
Then pass that interface around instead.

Reply:
Unfortunately we can't make existing classes implement interfaces after their definition in Kotlin, like we do in Swift :(. 

A possible alternative that might be better than this is to encapsulate the application context inside the `VariableDataProvider` and pass that around instead of the application context itself... We would still be passing the context around, but it would be masked, which seems a bit better. I will do that for now.

Reply:
Also note that we will also need the context to access the application's `R` strings in the other variable data functions.

Reply:
Made the changes I suggested here: https://github.com/RevenueCat/purchases-android/pull/1251/commits/b02017f784194c769bcdd764996896f9a3fae2a1. Lmk what you think!

Reply:
> Also note that we will also need the context to access the application's R strings in the other variable data functions.

Yeah so the indirection might make things even harder huh? Okay ignore me then!